### PR TITLE
Allow selecting synchronization strategy

### PR DESCRIPTION
**Pull request by FelonEkonom**
Relates to https://github.com/membraneframework/membrane_core/issues/734

### CODE REVIEW

File: lib/membrane/timestamp_queue.ex [hunk 1]
```diff
@@ -37,30 +37,32 @@ defmodule Membrane.TimestampQueue do
             metric_unit: :buffers | :bytes | :time,
             pad_queues: %{optional(Pad.ref()) => pad_queue()},
             pads_heap: Heap.t(),
-            registered_pads: MapSet.t(),
+            blocking_registered_pads: MapSet.t(),
+            registered_pads_offsets: %{optional(Pad.ref()) => integer()},
             awaiting_pads: [Pad.ref()],
```
Comment (mat-hek) [inline lib/membrane/timestamp_queue.ex:43]: let's add a comment that it contains at most one element

File: lib/membrane/timestamp_queue.ex [hunk 2]
```diff
@@ -267,8 +329,18 @@ defmodule Membrane.TimestampQueue do
...
+          case timestamp_queue.synchronization_strategy do
+            :synchronize_on_arrival ->
+              -timestamp_queue.current_queue_time
+
+            :explicit_offsets ->
+              offset = Map.get(timestamp_queue.registered_pads_offsets, pad_ref, 0)
+              offset - (buffer.dts || buffer.pts)
+          end
+
+        push_pad_on_heap(timestamp_queue, pad_ref, priority)
```
Comment (mat-hek) [inline lib/membrane/timestamp_queue.ex]: As we spoke, let's move that to push_buffer

File: lib/membrane/timestamp_queue.ex [hunk 3]
```diff
@@ -144,8 +199,14 @@ defmodule Membrane.TimestampQueue do
...
-            nil -> timestamp_queue.current_queue_time - (buffer.dts || buffer.pts)
-            valid_offset -> valid_offset
+            nil when timestamp_queue.synchronization_strategy == :synchronize_on_arrival ->
+              (buffer.dts || buffer.pts) - timestamp_queue.current_queue_time
+
+            nil when timestamp_queue.synchronization_strategy == :explicit_offsets ->
+              Map.get(timestamp_queue.registered_pads_offsets, pad_ref, 0)
+
+            valid_offset ->
+              valid_offset
```
Comment (mat-hek) [inline lib/membrane/timestamp_queue.ex]: let's handle the first buffer separately

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: mix.exs [hunk 4]
```diff
@@ -38,7 +38,7 @@ defmodule Membrane.TimestampQueue.Mixfile do
   defp deps do
     [
       {:membrane_core, "~> 1.0"},
-      {:heap, "~> 3.0"},
+      {:heap, "~> 2.0"},
```
Comment (mat-hek) [inline mix.exs:41]: why this?

Comment (FelonEkonom) [reply mix.exs:41]: Because `3.0` has weird licence

**Review by mat-hek — APPROVED** [review]

**Pull request merged**